src/libs/fullstory/index.native.ts

```diff
@@ -19,37 +18,32 @@ const FS = {
     consent: (c: boolean) => FullStory.consent(c),
 
     /**
-     * Initializes the FullStory session with the provided session information.
+     * Initializes the FullStory metadata with the provided metadata information.
      */
-    consentAndIdentify: (value: OnyxEntry<Session>) => {
+    consentAndIdentify: (value: OnyxEntry<UserMetadata>) => {
         try {
-            const session: UserSession = {
-                email: value?.email,
-                accountID: value?.accountID,
-            };
-            // set consent
+            // We only use FullStory in production environment
             FullStory.consent(true);
-            FS.fsIdentify(session);
+            FS.fsIdentify(value);
+            
         } catch (e) {
             // error handler
         }
     },
 
     /**
-     * Sets the FullStory user identity based on the provided session information.
-     * If the session is null or the email is 'undefined', the user identity is anonymized.
-     * If the session contains an email, the user identity is defined with the email and account ID.
+     * Sets the FullStory user identity based on the provided metadata information.
+     * If the metadata is null or the email is 'undefined', the user identity is anonymized.
+     * If the metadata contains an accountID, the user identity is defined with it.
      */
-    fsIdentify: (session: UserSession) => {
-        if (!session || session.email === 'undefined') {
-            // anonymize FullStory user identity session
+    fsIdentify: (metadata: UserMetadata | null) => {
+        if (!metadata?.accountID) {
```

Comment:
Should we also check that the `metadata` is defined and return early if not? especially given we are not deployed to prod yet

Reply:
I think that the `metadata?` is kind of doing that. When I tried `(!metadata || !metadata.accountID)`, typescript made me rewrite it this way

Reply:
cool, I see it now, I havent read properly its negated